File: web/src/model/victron_controller/dashboard/v0_2_0/from_0_1_0_forecasts.ts
// @ts-nocheck
import {Forecasts as v0_1_0_Forecasts} from '../v0_1_0/Forecasts'
import {Forecasts as v0_2_0_Forecasts} from './Forecasts'

export function convert__forecasts__from__0_1_0(from: v0_1_0_Forecasts): v0_2_0_Forecasts {
    return new v0_2_0_Forecasts (
        JSON.parse(JSON.stringify(from.solcast)),
        JSON.parse(JSON.stringify(from.forecast_solar)),
        JSON.parse(JSON.stringify(from.open_meteo))
    )
}

File: crates/dashboard-model/src/victron_controller/dashboard/v0_2_0/from_0_1_0_forecasts.rs


pub fn convert__forecasts__from__0_1_0(from: &crate::victron_controller::dashboard::v0_1_0::forecasts::Forecasts) -> crate::victron_controller::dashboard::v0_2_0::forecasts::Forecasts {
    crate::victron_controller::dashboard::v0_2_0::forecasts::Forecasts {
        solcast: serde_json::from_value(serde_json::to_value(&from.solcast).unwrap()).unwrap(),
        forecast_solar: serde_json::from_value(serde_json::to_value(&from.forecast_solar).unwrap()).unwrap(),
        open_meteo: serde_json::from_value(serde_json::to_value(&from.open_meteo).unwrap()).unwrap(),
    }
}